Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -8867,6 +8867,12 @@ void WebPageProxy::revokeGeolocationAuthorizationToken(const String& authorizati
 void WebPageProxy::queryPermission(const ClientOrigin& clientOrigin, const PermissionDescriptor& descriptor, CompletionHandler<void(std::optional<PermissionState>)>&& completionHandler)
 {
     bool canAPISucceed = true;
+
+    // FIXME: Current behavior always changes denied to prompt (except in the case of notifications). 
```

Comment:
It's a bit too long and not in the right place... Let's just say "FIXME: we should set shouldChangeDeniedToPrompt after first permission request like we do for notifications." and put it under `name = "geolocation"_s;`.